packages/core/src/agent/index.ts

```diff
@@ -326,56 +326,58 @@ export class Agent<
 
           await memory.saveMessages({
             memoryConfig,
-            messages: responseMessagesWithoutIncompleteToolCalls.map((message: CoreMessage | CoreAssistantMessage) => {
-              const messageId = randomUUID();
-              let toolCallIds: string[] | undefined;
-              let toolCallArgs: Record<string, unknown>[] | undefined;
-              let toolNames: string[] | undefined;
-              let type: 'text' | 'tool-call' | 'tool-result' = 'text';
-
-              if (message.role === 'tool') {
-                toolCallIds = (message as CoreToolMessage).content.map(content => content.toolCallId);
-                type = 'tool-result';
-              }
-              if (message.role === 'assistant') {
-                const assistantContent = (message as CoreAssistantMessage).content as Array<TextPart | ToolCallPart>;
-
-                const assistantToolCalls = assistantContent
-                  .map(content => {
-                    if (content.type === 'tool-call') {
-                      return {
-                        toolCallId: content.toolCallId,
-                        toolArgs: content.args,
-                        toolName: content.toolName,
-                      };
-                    }
-                    return undefined;
-                  })
-                  ?.filter(Boolean) as Array<{
-                  toolCallId: string;
-                  toolArgs: Record<string, unknown>;
-                  toolName: string;
-                }>;
-
-                toolCallIds = assistantToolCalls?.map(toolCall => toolCall.toolCallId);
-
-                toolCallArgs = assistantToolCalls?.map(toolCall => toolCall.toolArgs);
-                toolNames = assistantToolCalls?.map(toolCall => toolCall.toolName);
-                type = assistantContent?.[0]?.type as 'text' | 'tool-call' | 'tool-result';
-              }
+            messages: responseMessagesWithoutIncompleteToolCalls.map(
+              (message: CoreMessage | CoreAssistantMessage, index) => {
```

Comment:
huge diff here cause I added `index` to this line which changed indenting for everything